modules/nf-core/plink/recode/main.nf

```diff
@@ -0,0 +1,59 @@
+process PLINK_RECODE {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::plink=1.90b6.21" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/plink:1.90b6.21--h779adbc_1':
+        'quay.io/biocontainers/plink:1.90b6.21--h779adbc_1' }"
+
+    input:
+    tuple val(meta), path(bed), path(bim), path(fam)
+
+    output:
+    tuple val(meta), path("*.ped")                    , optional:true, emit: ped
+    tuple val(meta), path("*.map")                    , optional:true, emit: map
+    tuple val(meta), path("*.txt")                    , optional:true, emit: txt
+    tuple val(meta), path("*.raw")                    , optional:true, emit: raw
+    tuple val(meta), path("*.traw")                   , optional:true, emit: traw
+    tuple val(meta), path("*.beagle.dat")             , optional:true, emit: beagledat
+    tuple val(meta), path("*.chr-*.dat")              , optional:true, emit: chrdat
+    tuple val(meta), path(".*chr-*.map")              , optional:true, emit: chrmap
+    tuple val(meta), path("*.recode.geno.txt")        , optional:true, emit: geno
+    tuple val(meta), path("*.recode.pheno.txt")       , optional:true, emit: pheno
+    tuple val(meta), path("*.recode.pos.txt")         , optional:true, emit: pos
+    tuple val(meta), path("*.recode.phase.inp")       , optional:true, emit: phase
+    tuple val(meta), path("*.info")                   , optional:true, emit: info
+    tuple val(meta), path("*.lgen")                   , optional:true, emit: lgen
+    tuple val(meta), path("*.list")                   , optional:true, emit: list
+    tuple val(meta), path("*.gen")                    , optional:true, emit: gen
+    tuple val(meta), path("*.gen.gz")                 , optional:true, emit: gengz
+    tuple val(meta), path("*.sample")                 , optional:true, emit: sample
+    tuple val(meta), path("*.rlist")                  , optional:true, emit: rlist
+    tuple val(meta), path("*.strct_in")               , optional:true, emit: strctin
+    tuple val(meta), path("*.tped")                   , optional:true, emit: tped
+    tuple val(meta), path("*.tfam")                   , optional:true, emit: tfam
+    tuple val(meta), path("*.vcf")                    , optional:true, emit: vcf
+    tuple val(meta), path("*.vcf.gz")                 , optional:true, emit: vcfgz
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    plink \\
+        --bfile ${meta.id} \\
```

Comment:
Why do you use `meta.id` here instead of `prefix`? :) (Just curious what this argument does, for most cases it is preffered to use `prefix` for the greater flexibility)

Reply:
Good question. In order to make PLINK_RECODE similar/consistent to previous PLINK modules (such as [plink_extract](https://github.com/nf-core/modules/blob/master/modules/nf-core/plink/extract/main.nf)) I used `${meta.id}`. I am not aware about the benefits of using prefix, though, would you suggest changing it?

Reply:
What does the `-bfile` argument do exactly?

Reply:
As per PLINK's documentation, "the` --bfile` flag causes the binary fileset plink.bed + plink.bim + plink.fam to be referenced.  If a prefix is given, it replaces all instances of 'plink'.". So tipically you add here the name/prefix of the these three files.

Reply:
Ah! Okay, maybe you should specify this in the docs that the three input files need to have the same prefix. You could also automatically detect the name of these files and use this as input. `meta.id` won't always be the same as the prefix of the input files.
`bed.baseName` should do the trick in getting the name :)

Reply:
I was reading the reference and there is also: " `--bed`, `--bim`, and `--fam` let you specify the full name of one part of the PLINK 1 binary fileset, taking precedence over `--bfile`. "  I am going to think a bit more about how to handle this, then.

Reply:
Ah I think it would be better if you can specify them separately :)

Reply:
Thank you for the suggestion! I addresed it here: https://github.com/nf-core/modules/pull/2683/commits/5c413babb4ea7d3af8390d5ecaacc051e6e60cf0 where each input is now specified separately.

Reply:
Are these files different for each sample? If so I would leave the input channel as it was since they will always be linked to eachother that way (based on how the channel logic works in Nextflow)

Reply:
That's true. I haven't thought about that. So, if I understand correctly, the channel can be one, composed of a 'single' tuple but then I can call the elements of the channel individually in the script section. So instead of: 

```
    input:
    tuple val(meta), path(bed)
    tuple val(meta), path(bim)
    tuple val(meta), path(fam)
    
    script:
     """
    plink \\
        --bed ${bed}  \\
        --bim ${bim}  \\
        --fam ${fam}  \\ [etc]
      """
```
It is better to:

```
    input:
    tuple val(meta), path(bed),  path(bim), path(fam)
    
    script:
    """
    plink \\
        --bed ${bed}  \\
        --bim ${bim}  \\
        --fam ${fam}  \\ [etc]
   """
```

Reply:
Yep that's correct!!!

Reply:
Here it is https://github.com/nf-core/modules/pull/2683/commits/50a3d4f1c2e6ed5c0819f4bc9d1769ee29183906 :)